opm/input/eclipse/Schedule/Group/GSatProd.hpp

```diff
@@ -25,22 +25,27 @@
 #include <map>
 #include <string>
 
+#include <opm/input/eclipse/Deck/UDAValue.hpp>
+#include <opm/input/eclipse/Units/UnitSystem.hpp>
+
 namespace Opm {
 
+    class SummaryState;
+
     /// Group level satellite production
     class GSatProd
     {
     public:
         /// Satellite production rates for a single group.
         struct GSatProdGroup
         {
-            /// Satellite production rate items
-            enum Rate { Oil, Gas, Water, Resv, GLift, };
-
-            /// Satellite production rates.
-            ///
-            /// One rate for each enumerated item.
-            std::array<double, 5> rate{};
+            UDAValue oil_rate;
+            UDAValue gas_rate;
+            UDAValue water_rate;
+            double resv_rate;
+            double glift_rate;
```

Comment:
If we mark items 5, 6, and 7 as `UDA`, then I think we should continue to use an `array<>` for these in order to simplify serialisation.